From cb9204f1aed6635af8977f20650105e92a8d195d Mon Sep 17 00:00:00 2001
From: Marc Mutz <marc.mutz@qt.io>
Date: Fri, 26 Sep 2025 10:12:28 +0200
Subject: [PATCH 220/553] Mark QStringBuilder as security-critical

As an algorithm on QString / QByteArray, it's a priori
security-critical.

The situation may be different if QStringBuilder was implemented only
using (public) QString / QByteArray API. Then one could argue that
QString should firewall the component from security risks, but it's
not, so QString doesn't.

The implementation is mostly in the header, and while the
implementation in the .cpp file is _currently_ trivial, looking a bit
ahead, the goal should, as always, be to move more stuff from the
header out-of-line, so proactively keep the .cpp file at critical,
too.

Task-number: QTBUG-135195
Pick-to: 6.8
Change-Id: I5c28f677e74632130d4b79159e8eafd9686c7821
Reviewed-by: Ivan Solovev <ivan.solovev@qt.io>
(cherry picked from commit 79cefaba8bb124d74795c943edc817b04c4815eb)
Reviewed-by: Qt Cherry-pick Bot <cherrypick_bot@qt-project.org>
---
 src/corelib/text/qstringbuilder.cpp | 1 +
 src/corelib/text/qstringbuilder.h   | 1 +
 2 files changed, 2 insertions(+)

diff --git a/src/corelib/text/qstringbuilder.cpp b/src/corelib/text/qstringbuilder.cpp
index 738ce833ef2..b1505259b69 100644
--- a/src/corelib/text/qstringbuilder.cpp
+++ b/src/corelib/text/qstringbuilder.cpp
@@ -1,5 +1,6 @@
 // Copyright (C) 2016 The Qt Company Ltd.
 // SPDX-License-Identifier: LicenseRef-Qt-Commercial OR LGPL-3.0-only OR GPL-2.0-only OR GPL-3.0-only
+// Qt-Security score:critical reason:data-parser
 
 #include "qstringbuilder.h"
 #include <private/qstringconverter_p.h>
diff --git a/src/corelib/text/qstringbuilder.h b/src/corelib/text/qstringbuilder.h
index 0b296d5142a..2abe594198b 100644
--- a/src/corelib/text/qstringbuilder.h
+++ b/src/corelib/text/qstringbuilder.h
@@ -1,5 +1,6 @@
 // Copyright (C) 2020 The Qt Company Ltd.
 // SPDX-License-Identifier: LicenseRef-Qt-Commercial OR LGPL-3.0-only OR GPL-2.0-only OR GPL-3.0-only
+// Qt-Security score:critical reason:data-parser
 
 #include <QtCore/qstring.h>
 
-- 
2.51.2

